src/graph/SysInfoHandler.cpp

```diff
@@ -0,0 +1,83 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "fs/FileUtils.h"
+#include "graph/SysInfoHandler.h"
+#include "process/ProcessUtils.h"
+#include <proxygen/httpserver/RequestHandler.h>
+#include <proxygen/lib/http/ProxygenErrorEnum.h>
+#include <proxygen/httpserver/ResponseBuilder.h>
+
+namespace nebula {
+namespace graph {
+
+using nebula::ProcessUtils;
+using proxygen::HTTPMessage;
+using proxygen::HTTPMethod;
+using proxygen::ProxygenError;
+using proxygen::UpgradeProtocol;
+using proxygen::ResponseBuilder;
+
+void SysInfoHandler::onRequest(std::unique_ptr<HTTPMessage> headers) noexcept {
+    if (headers->getMethod().value() != HTTPMethod::GET) {
+        // Unsupported method
+        err_ = HttpCode::E_UNSUPPORTED_METHOD;
+        return;
+    }
+}
+
+void SysInfoHandler::onBody(std::unique_ptr<folly::IOBuf>) noexcept {
+    // Do nothing, we only support GET
+}
+
+void SysInfoHandler::onEOM() noexcept {
+    switch (err_) {
+        case HttpCode::E_UNSUPPORTED_METHOD:
+            ResponseBuilder(downstream_)
+                .status(WebServiceUtils::to(HttpStatusCode::METHOD_NOT_ALLOWED),
+                        WebServiceUtils::toString(HttpStatusCode::METHOD_NOT_ALLOWED))
+                .sendWithEOM();
+            return;
+        default:
+            break;
+    }
+
+    folly::dynamic vals = getSysInfo();
+    ResponseBuilder(downstream_)
+        .status(WebServiceUtils::to(HttpStatusCode::OK),
+                WebServiceUtils::toString(HttpStatusCode::OK))
+        .body(folly::toJson(vals))
+        .sendWithEOM();
+}
+
+
+void SysInfoHandler::onUpgrade(UpgradeProtocol) noexcept {
+    // Do nothing
+}
+
+
+void SysInfoHandler::requestComplete() noexcept {
+    delete this;
+}
+
+void SysInfoHandler::onError(ProxygenError error) noexcept {
+    LOG(ERROR) << "Web service StorageHttpHandler got error: "
+               << proxygen::getErrorString(error);
+}
+
+folly::dynamic SysInfoHandler::getSysInfo() {
+    folly::dynamic json = folly::dynamic::object();
+    auto memUsage = ProcessUtils::getMemUsage();
+    json["memory_usage"] = memUsage;
+
+    auto cpuUsage = ProcessUtils::getCpuUsage();
+    json["cpu_usage"] = cpuUsage;
+
+    return json;
+}
```

Comment:
_**System Info**_ involves many metrics. What's your plan on the design of this interface? For example, what if we want to separate the CPU usage into  _user_, _sys_, _si_ metrics in future?

Reply:
I Just want this interface act as `docker stats`. Via this this interface, we can briefly get cpu/memory/disk consumption. We should let users collect the more detail metrics via third party tools.In the future, we can use kubernetes manage nebula, At that time, We can do more and collect more metrics via prometheus.  Since nebula 1.0 not work very well on kubernetes, I think it's better support the minimum feature set to help users build their monitor system as easy as possible.